single-function/valid-parentheses/submission.js

```diff
@@ -0,0 +1,48 @@
+const isValidParentheses = (str) => {
+  if(str.length <= 1) { return false }
+  const parenMap = { '{':'}', '(':')', '[':']' };
+  const stack = [];
+  for (let i = 0; i < str.length; i++) {
+    if (parenMap[str[i]]) {
+      stack.push(parenMap[str[i]]);
+    } else if (stack.pop() !== str[i]) {
+      return false;
+    }
+  }
+  return (str.length % 2 !== 0) || (stack.length !== 0) || true;
+};
+
+/*
+Input: '()()()'
+Output: boolean
+Constraints: n/a
+
+Strategy: Step through the given string, if a valid opening paren push closing paren on to stack.
+If a non valid opening paren found pop from the stack and not equal to the current string value invalidate string.
+
+Transformation Steps:
```

Comment:
Let's call this Transformation Example(s) so it encourages showing more than one to cover multiple cases. Here, another useful case to illustrate would be having too many open parens.